src/main/java/uk/gov/hmcts/reform/roleassignment/controller/advice/RoleAssignmentControllerAdvice.java

```diff
@@ -55,10 +60,53 @@
             ex,
             BAD_REQUEST,
             BAD_REQUEST.value(),
-            "Bad Request"
+            Constants.BAD_REQUEST
         );
     }
 
+    @ExceptionHandler(HttpMessageNotReadableException.class)
+    public ResponseEntity<ErrorResponse> notReadableException(final HttpMessageNotReadableException e) {
+        return deserializeError(e, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(NullPointerException.class)
+    public ResponseEntity<ErrorResponse> nullException(final NullPointerException e) {
+        return nullError(HttpStatus.BAD_REQUEST);
+    }
+
+    private ResponseEntity<ErrorResponse> deserializeError(final Exception exception, final HttpStatus httpStatus) {
+        String cause = exception.getMessage();
+        if (cause.contains("RoleType")) {
+            return new ResponseEntity<>(
+                ErrorResponse.builder()
+                    .errorCode(400)
+                    .errorDescription("The Role Type parameter is not valid")
+                    .errorMessage(Constants.BAD_REQUEST).build(), httpStatus);
+        } else if (cause.contains("UUID")) {
+            String inputStart = StringUtils.substringAfter(cause, "\"");
+            String inputEnd = StringUtils.substringBefore(inputStart, "\"");
+            try {
+                ValidationUtil.validateInputParams(UUID_PATTERN, inputEnd);
+            } catch (Exception e) {
+                return new ResponseEntity<>(
+                    ErrorResponse.builder()
+                        .errorCode(400).errorDescription(e.getMessage() + ": UUID")
+                        .errorMessage(Constants.BAD_REQUEST).build(), httpStatus);
+            }
+        }
+        return new ResponseEntity<>(ErrorResponse.builder().errorCode(400).errorDescription(exception.getMessage())
+                                        .errorMessage(Constants.BAD_REQUEST).build(), httpStatus);
```

Comment:
Check if its a roletype or uuid type of error. If not, just pass the original error down for safety.